### PR TITLE
Milestone 2: Decorator

### PR DESCRIPTION
**Pull request by Profsain**
I completed this part of the project by implementing the following requirement
- Created decorator interface class with correct_name method
- Created decorator class
- Created CapitalizeDecorator class to capitalize name 
- Created TrimmerDecroator class to remove whitespace
- Tested decorator action

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
```
Comment (cynthiainga) [inline decorator.rb]: - [ ] Please kindly add this part to their file `capitalize_decorator` to have a clean code and for best practice 🙃🙃

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.strip
+  end
+end
```
Comment (cynthiainga) [inline decorator.rb]: - [ ] Also kindly add this part in their file `timmer_decorator`  to have a clean code and for best practice 🙃 😉

File: decorator.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.strip
```
Comment (cynthiainga) [inline decorator.rb]: - [ ] As in the requirement, you should have a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters
eg:
````ruby
    def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
````

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @Profsain, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- [ ] Use correct gitflow

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇 👇


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorator.rb (hunk 3 above)
Comment (Profsain) [reply decorator.rb]: Thank you for the feedback. I will implement the changes.

**Review by ShahierNashaat — APPROVED** [review]
Hi @Profsain, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**